test/highlight/comments.lua

```diff
@@ -0,0 +1,28 @@
+-- comment
+
+-- first line \
+second line
+
+-- first line trailing ws \    
+   second line
+
+--[[ block comment ]]
+--[[ block comment on next line ]]
+
+-- comment line
+noncomment;
+
+-- a
+-- b
+-- c
+
+--[[
+multiple
+lines
+of
+comment
+]]
+
+--[==[ block comment with equals ]==]
+
+--[[ Unterminated block comment
```

Comment:
Please make sure that all files have trailing newlines. Otherwise someone else (probably me) is just going to pollute the commit history with such whitespace changes.